src/libPMacc/include/memory/buffers/HostBufferIntern.hpp

```diff
@@ -70,9 +76,9 @@ class HostBufferIntern : public HostBuffer<TYPE, DIM>
     {
         __startOperation(ITask::TASK_HOST);
 
-        if (pointer && ownPointer)
+        if (this->getPointer() && ownsPointer)
         {
-            CUDA_CHECK(cudaFreeHost(pointer));
+            CUDA_CHECK(cudaFreeHost(this->getPointer()));
         }
```

Comment:
Please set both pointer `pointer` and `basePointer` to `NULL` after the if condition, this helps to detect if already freed memory is used.